### ISSUE TITLE
Divergence at IRR values near zero

### ISSUE THREAD

**Issue by aziadeh** [better/irr#1]
Hey,

Really like how fast the algorithm is but at IRR values that are supposed to be close to zero, the irr_binary_search calculation diverges, irr_newton seem to find the proper solution as well as numpy.irr. 

Here's an example of a list that would return a -1 instead of -.01: 
`[-13897.515699392789, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, 678.69417667002108, -426.40303828879155]
`
You've ever ran into such problems?

**Comment by erikbern** [comment]
interesting – will take a look

**Comment by aziadeh** [comment]
I have a few more examples if you need them.

**Comment by erikbern** [comment]
sure – if you can send them over i'll add them as unit tests

**Comment by aziadeh** [comment]
`[-12138.436076306429, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, 576.20775733699475, -528.88945762181788]
`


`[-10351.121144852736, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, 450.71546738230256, -654.38174757651018]
`

`[-15634.416942708685, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, 800.12182531566373, -304.9753896431489]
`
There's an obvious pattern. I still have more but not sure they'll add much because the structure is more or less the same ! Thanks btw:)

**Comment by erikbern** [comment]
I think the problem is your polynomials have two roots, one where you expect it, and then another weird root for some large negative number. let me figure it out later

**Comment by erikbern** [comment]
Yeah here's the plot of the residual as a function of rate – note that it has two zeroes:

![image](https://user-images.githubusercontent.com/1027979/32232593-c45b555a-be2e-11e7-84a5-1c9a12160ab1.png)

Note sure why it doesn't converge to the other zero though
